client/src/modules/dashboard/shared/components/Layout.tsx

```diff
@@ -2,12 +2,17 @@ import { HStack } from '@chakra-ui/layout';
 import { LinkButton } from 'chakra-next-link';
 import { useRouter } from 'next/router';
 import React from 'react';
+import { useCheckPermission } from 'hooks/useCheckPermission';
 
 const links = [
   { text: 'Chapters', link: '/dashboard/chapters' },
   { text: 'Events', link: '/dashboard/events' },
   { text: 'Venues', link: '/dashboard/venues' },
-  { text: 'Sponsors', link: '/dashboard/sponsors' },
+  {
+    text: 'Sponsors',
+    link: '/dashboard/sponsors',
+    permission: 'sponsors-manage',
```

Comment:
```suggestion
    requiredPermission: 'sponsors-manage',
```
It's a bit verbose, but I think it's clearer.  

Also, I think it would be good to share `Permission` between the client and the server. That's a little involved and out of scope, though, so I'll take care of that separately.